dvc/analytics/system_info.py

```diff
@@ -0,0 +1,29 @@
+import platform
+import sys
+import distro
+
+
+def collect():
+    system = platform.system()
+
+    if system == "Windows":
+        version = sys.getwindowsversion()
+
+        return {
+            "os": "windows",
+            "windows_version_build": version.build,
+            "windows_version_major": version.major,
+            "windows_version_minor": version.minor,
+            "windows_version_service_pack": version.service_pack,
+        }
+
+    if system == "Darwing":
+        return {"os": "mac", "mac_version": platform.mac_ver()[0]}
+
+    if system == "Linux":
+        return {
+            "os": "linux",
+            "linux_distro": distro.id(),
+            "linux_distro_like": distro.like(),
+            "linux_distro_version": distro.version(),
+        }
```

Comment:
No fallback if system is not the one that we expect. Older implementation at least raised NotImplementedError.

Reply:
True, @efiop , the previous implementation relied on `getattr`, I missed that one ^